### PR TITLE
fix(SonarQube): Fixes inspired by SonarQube evaluation

### PR DESCRIPTION
**Pull request by ygworldr**
### For developers

Fixes inspired by [SonarQube](https://docs.sonarqube.org/latest/setup/get-started-2-minutes/).

1. The cyclomatic complexity one was highlighted by my linters and indeed needed to be acted upon.
2. The rest seem… A little nitpicky. 

## FlowFast

[FlowFast card link](https://worldr.flowfast.io/6183698).

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ tasks:
             - coverage run -m pytest --doctest-modules
             - coverage combine
             - coverage report -m
+            - coverage xml
```
Comment (ygworldr) [inline Taskfile.yml:33]: This is so SonarQube can display the test coverage. Whatever, it does no harm to have it.

File: setupr/console.py [hunk 2]
```diff
@@ -196,10 +198,10 @@ class MutuallyExclusiveOption(click.Option):
     def __init__(self, *args, **kwargs):  # type: ignore
         """Initialize."""
         self.mutually_exclusive = set(kwargs.pop("mutually_exclusive", []))
-        help = kwargs.get("help", "")
+        _help = kwargs.get("help", "")
```
Comment (ygworldr) [inline setupr/console.py:201]: This serious bug overshadowed a built-in variable.

File: setupr/downloader.py [hunk 3]
```diff
@@ -267,11 +267,11 @@ def execute_script(  # noqa
                 for line in err.splitlines():
                     console.log(f"[{COLOUR_FAIL}]final script stderr: {line}")
                     rlog.error("stderr", script=script, line=line)
-            returnCode = proc.returncode
-            if returnCode != 0:
-                wprint(f"Exit Code {returnCode}", level="failure")
-                rlog.error("Return Code", script=script, code=returnCode)
+            return_code = proc.returncode
```
Comment (ygworldr) [inline setupr/downloader.py:270]: Wrong format of a variable name.

File: tests/test_downloader.py [hunk 4]
```diff
@@ -14,14 +14,16 @@
...
 import requests_mock
-
-# import requests_mock
 from pendulum.parser import parse
 from pendulum.parsing.exceptions import ParserError
 from rich.progress import Progress, TaskID
 
 from setupr.downloader import Downloader, copy_url, download, take_backup
 
+URL = "https://worldr.com/index.html"
```
Comment (ygworldr) [inline tests/test_downloader.py:23]: A string used several times should be a constant.

File: tests/test_downloader.py [hunk 5]
```diff
@@ -30,27 +32,25 @@ def test_copy_url(mocked_done_event: Mock, is_set: bool) -> None:
...
         progress = MagicMock(spec=Progress)
-        copy_url(MagicMock(spec=TaskID), url, dst, progress)
+        copy_url(MagicMock(spec=TaskID), URL, dst, progress)
         assert progress.start_task.called
         assert progress.update.called
 
 
 @patch("setupr.downloader.done_event")
-def test_copy_url_RequestException(mocked_done_event: Mock) -> None:
+def test_copy_url_request_exception(mocked_done_event: Mock) -> None:
```
Comment (ygworldr) [inline tests/test_downloader.py:44]: Bad file name.

File: tests/test_downloader.py [hunk 6]
```diff
@@ -96,31 +95,30 @@ def test_download_failed() -> None:
...
+            ANY, ANY, URL, tmpdirname + INDEX, ANY
         )
         assert fut.result.called
 
 
 def test_take_backup() -> None:
     with tempfile.TemporaryDirectory(prefix="setupr_tests_") as tmpdirname:
-        file = pathlib.Path(tmpdirname, "test.txt")
-        with open(file, "w") as fp:
+        _file = pathlib.Path(tmpdirname, "test.txt")
```
Comment (ygworldr) [inline tests/test_downloader.py:114]: Overshadowed a built-in.

File: tests/test_downloader.py [hunk 7]
```diff
@@ -96,31 +95,30 @@ def test_download_failed() -> None:
...
+        with open(_file, "w") as fp:
             fp.write("created temporary file 0\n")
-        assert take_backup(file) == file
+        assert take_backup(_file) == _file
         all_files = glob(f"{tmpdirname}/archives/test*.txt")
         assert len(all_files) == 1, "There should be only one…"
         date_string = re.findall(
-            r"_[0-9]{4}-[0-9]{2}-[0-9]{2}T.*\.",
+            r"_\d{4}-\d{2}-\d{2}T.*\.",
```
Comment (ygworldr) [inline tests/test_downloader.py:121]: Simpler regexp.

File: tests/test_downloader.py [hunk 8]
```diff
@@ -279,8 +280,8 @@ def test_execute_script(
         mocked_confirm.ask = MagicMock(return_value=True)
         with patch("setupr.downloader.Console") as mocked_console:
             downloader._gpg.validate_worldr_signature = Mock(return_value=True)
-            mConsole = MagicMock()
-            mocked_console.return_value = mConsole
+            m_console = MagicMock()
```
Comment (ygworldr) [inline tests/test_downloader.py:283]: Wrong variable name.

File: tests/test_downloader_functional.py [hunk 9]
```diff
@@ -33,7 +33,7 @@ def test_get_real_scripts(what: str, version: str, basename: str) -> None:
     """
     # Check if we can connect to the Internet.
     try:
-        requests.head("http://www.google.com/", timeout=1)
+        requests.head("https://www.google.com/", timeout=1)
```
Comment (ygworldr) [inline tests/test_downloader_functional.py:36]: Insecurely using `http` instead of `https`.

File: tests/test_downloader_functional.py [hunk 10]
```diff
@@ -92,7 +92,7 @@ def test_fetch_goss(file: str, url: str) -> None:
...
```
Comment (ygworldr) [inline tests/test_downloader_functional.py:95]: Insecurely using `http` instead of `https`.

File: tests/test_gpg.py [hunk 11]
```diff
@@ -9,6 +9,8 @@
 
 from setupr.gpg import GPG
 
+CHARON_LORD_DUNSANY_TXT = "charon-lord-dunsany.txt"
```
Comment (ygworldr) [inline tests/test_gpg.py:12]: String used in several places should be constant.

File: tests/test_pre_flight.py [hunk 12]
```diff
@@ -65,7 +65,7 @@ def _side_efect(cmd):  # pragma: no cover
         assert preflight._downloader.fetch.called
 
 
-def test_PreFlight_goss_is_cached(preflight):
+def test_pre_flight_goss_is_cached(preflight):
```
Comment (ygworldr) [inline tests/test_pre_flight.py:68]: Bad function name.

File: tests/test_pre_flight.py [hunk 13]
```diff
@@ -130,7 +130,7 @@ def test_run(retcode, mock_preflight):
 
 @patch("setupr.pre_flight.PreFlight.goss", new_callable=PropertyMock)
 @patch("setupr.pre_flight.take_backup")
-def test_run_ProcessExecutionError(m_take_backup, mock_goss, mock_preflight):
+def test_run_process_execution_error(m_take_backup, mock_goss, mock_preflight):
```
Comment (ygworldr) [inline tests/test_pre_flight.py:133]: Bad function name.

**Review by ygworldr — COMMENTED** [review]
Some reasons why the code was changed…

**Comment by ygworldr** [comment]
Anyone is welcome to have a look…

**Review by Kryvchun — APPROVED** [review]
Thanks for comments

File: setupr/console.py [hunk 14]
```diff
@@ -77,6 +77,8 @@ class VerboseFilter(logging.Filter):
         """Filter log entries on verbose flag."""
 
         def __init__(self, param: str = "") -> None:
+            """Initialise."""
+            self.param = param
             self.param = param
```
Comment (agworldr) [inline setupr/console.py:82]: I don't know python well enough, but it looks like a duplication.

File: setupr/console.py [hunk 15]
```diff
@@ -325,22 +325,49 @@ def main(  # noqa: C901
     different for all the scripts. Please check the user documentation
     for the exact values.
     """
-    # Version
+    # Prints the current version and exit.
```
Comment (agworldr) [inline setupr/console.py:328]:
```suggestion
    # Prints the current version and exits.
```

File: setupr/console.py [hunk 16]
```diff
@@ -361,103 +388,101 @@ def main(  # noqa: C901
...
+        # should be able to not specify the option, so it could be None.
+        # However, then Path will raise because you cannot instanciate it
+        # with None. Urgh.
+        if service_account is None:
+            path = None
+        else:
+            path = Path(str(service_account))
+        data = InstallationData(service_account_json=path)
+        if data.fetch():
+            wprint("Got YMAL installation data.", level="info")
```
Comment (agworldr) [inline setupr/console.py:415]:
```suggestion
            wprint("Got YAML installation data.", level="info")
```

**Review by agworldr — APPROVED** [review]

File: setupr/console.py (hunk 14 above)
Comment (ygworldr) [reply setupr/console.py]: It is a duplicate! Thank you!

**Pull request merged**